packages/amp/amp.d.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * The namespace which overrides the built-in JSX namespace. It's necessary in
+ * order to use AMP components with typescript. Otherwise, TS will complain
+ * because it does not recognize the AMP components as valid HTML elements.
+ *
+ * @remarks
+ * For more information see:
+ * - https://github.com/Microsoft/TypeScript/issues/15449
+ * - https://stackoverflow.com/questions/50585952/typescript-and-google-amp-property-amp-img-does-not-exist-on-type-jsx-intrin.
+ */
+declare namespace JSX {
+  /**
+   * Override the built in JSX interface.
+   */
+  interface IntrinsicElements {
```

Comment:
Good question. We could use a workaround to add all AMP elements in the `@frontity/amp` package, something like:

```ts
declare global {
  namespace JSX {
    interface IntrinsicElements {
      [key: string]: any;
    }
  }
}
```

It would work because the types for the already-defined HTML elements are preserved.

In the [NextJS docs](https://nextjs.org/docs/advanced-features/amp-support/typescript), there's a link to a StackOverflow thread where that workaround is suggested. Also, they say that AMP has in their roadmap to add types for their elements, but it seems that's not going to happen in the near future (https://github.com/ampproject/amphtml/issues/13791#issuecomment-650460822).

Reply:
We can try that. See if it solves the problems on all packages.

@michalczaplinski what do you think? Do you agree? If so, could you refactor that?

Reply:
hmm, I'm not sure if adding 

```
interface IntrinsicElements {
  ...
  [key: string]: any;
}
```

would make sense here because then any string can be used as the JSX element and that makes some typos more likely. 

But more importantly, the `amp.d.ts` file is scoped to the AMP package so it would not solve the issue for all packages, only for the code that is in the AMP package.

If any other package would want to use that workaround, they would have to reference the types of AMP package like:

```
/// <reference types="@frontity/amp" />
```

This is because we are augumenting a namespace in the global scope and TS prevents it from leaking by default. It's the same reason why we have to e.g. reference the Cypress types in https://github.com/frontity/frontity/blob/ca55fa5d235ba297ed88f355459d63bcd2790f90/e2e/support/index.d.ts#L3-L3

So, in my opinion the best way would be to leave those types as they are, what do you think?

Reply:
Whatever works. If it can't be done or doesn't make sense, don't worry. It's just a matter of keeping everything as tidy and organized as possible 🙂

Reply:
ok, will keep it as it is in that case :)